### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by Reettaphant**
- FrontId is q part of the front edit route and changes as users select different fronts
- Restricts valid priorities when routing
- Adds a generic not found component.

### CODE REVIEW

File: public/client-v2/src/components/App.js [hunk 1]
```diff
@@ -20,8 +22,13 @@ const AppContainer = styled('div')`
 const App = () => (
   <AppContainer>
     <Switch>
-      <Route exact path="/:priority" component={FrontsEdit} />
+      <Route
+        exact
+        path={`/:priority(${Object.keys(priorities).join('|')})/:frontId?`}
```
Comment (RichieAHB) [inline public/client-v2/src/components/App.js]: Didn't know this exited for `react-router` 👍

File: public/client-v2/src/components/FrontsEdit/Fronts.js [hunk 2]
```diff
@@ -26,15 +30,23 @@ class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
     selectedFront: null
   };
 
+  componentWillMount() {
```
Comment (RichieAHB) [inline public/client-v2/src/components/FrontsEdit/Fronts.js]: These methods are being deprecated in future version of React.  For this one we can just set the state in the `constructor`. I actually don't think we should be using state here but actually just using `frontId` directly and using the prop to set the select value as it'll always be synced up no?

File: public/client-v2/src/components/FrontsEdit/Fronts.js [hunk 3]
```diff
@@ -26,9 +26,17 @@ type ComponentState = {
 };
 
 class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
-
   componentDidMount() {
-    this.props.frontsActions.getFrontsConfig();
+    this.props.frontsActions.getFrontsConfig().then(() => {
```
Comment (RichieAHB) [inline public/client-v2/src/components/FrontsEdit/Fronts.js]: I wonder whether we should handle this in the component above this? This seems like an odd side effect to have (a quick render then redirect when we had this information in the parent). Can we not check if the selected front is valid in the parent component then either render the selected front here or render the select to say "choose front". What do you think?

Comment (Reettaphant) [reply public/client-v2/src/components/FrontsEdit/Fronts.js]: Yes, I was thinking about it and it's a bit weird but we are fetching the fronts in this component. We could fetch the fronts in the component where we do the routing, we wouldn't need them for anything else except for deciding which routes are valid.

**Pull request merged**